internal/webhooks/runtime/extensionconfig_webhook.go

```diff
@@ -93,10 +101,121 @@ func (webhook *ExtensionConfig) validate(_ context.Context, _, _ *runtimev1.Exte
 			"can be set only if the RuntimeSDK feature flag is enabled",
 		)
 	}
+
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, validateExtensionConfigSpec(newExtensionConfig)...)
+
+	if len(allErrs) > 0 {
+		return apierrors.NewInvalid(runtimev1.GroupVersion.WithKind("ExtensionConfig").GroupKind(), newExtensionConfig.Name, allErrs)
+	}
 	return nil
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
 func (webhook *ExtensionConfig) ValidateDelete(_ context.Context, _ runtime.Object) error {
 	return nil
 }
+
+func validateExtensionConfigSpec(e *runtimev1.ExtensionConfig) field.ErrorList {
+	var allErrs field.ErrorList
+
+	specPath := field.NewPath("spec")
+
+	if e.Spec.ClientConfig.URL == nil && e.Spec.ClientConfig.Service == nil {
+		allErrs = append(allErrs, field.Required(
+			specPath.Child("clientConfig"),
+			"either URL or Service must be defined",
+		))
+	}
+	if e.Spec.ClientConfig.URL != nil && e.Spec.ClientConfig.Service != nil {
+		allErrs = append(allErrs, field.Forbidden(
+			specPath.Child("clientConfig"),
+			"only one of URL or Service can be defined",
+		))
+	}
+
+	// Validate URL
+	if e.Spec.ClientConfig.URL != nil {
+		if uri, err := url.ParseRequestURI(*e.Spec.ClientConfig.URL); err != nil {
+			allErrs = append(allErrs, field.Invalid(
+				specPath.Child("clientConfig", "url"),
+				*e.Spec.ClientConfig.URL,
+				"must be a valid URL, e.g. https://example.com",
+			))
+		} else if uri.Scheme != "http" && uri.Scheme != "https" {
+			allErrs = append(allErrs, field.Invalid(
+				specPath.Child("clientConfig", "url"),
+				*e.Spec.ClientConfig.URL,
+				"'https' and 'http' are the only allowed URL schemes, e.g. https://example.com",
+			))
+		}
+	}
+
+	// Validate Service if defined
+	if e.Spec.ClientConfig.Service != nil {
```

Comment:
Since we want to enforce https when using a service, how about adding a check to make sure that the `CABundle` field is not empty when `Service` is used?

Reply:
> Let's keep http support till we have more information on how to deal with cert-manager CAbundle injection. Issue ref: https://github.com/cert-manager/cert-manager/issues/5073

Addition: CABundle is not even enforced on mutating/validatingwebhooks. They just deny `http` for the url and always use  https.

Reply:
> Addition: CABundle is not even enforced on mutating/validatingwebhooks. They just deny http for the url and always use https.

General comment, we shouldn't necessarily align on upstream. I now a bunch of cases where the validation is not super great and the godocs sometimes don't exist at all :)